src/librustc_typeck/astconv.rs

```diff
@@ -2934,6 +2935,21 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         for ty in decl.inputs {
             visitor.visit_ty(ty);
         }
+        for predicate in predicates {
+            if let hir::WherePredicate::BoundPredicate(hir::WhereBoundPredicate {
+                bounded_ty,
+                bounds,
+                ..
+            }) = predicate
+            {
+                visitor.visit_ty(bounded_ty);
+                for bound in *bounds {
+                    if let hir::GenericBound::Trait(ptr, tbm) = bound {
+                        visitor.visit_poly_trait_ref(&ptr, *tbm);
+                    }
+                }
+            }
+        }
```

Comment:
Any reason not to use `walk_list!(visitor, walk_where_predicate, predicates);` here?

Reply:
Changed it to `walk_generics(&mut visitor, generics);`